From 3e3438c0a4e39698d0604b19be07b83aac7e41a8 Mon Sep 17 00:00:00 2001
From: apporc <appleorchard2000@gmail.com>
Date: Tue, 15 Mar 2016 11:55:45 +0800
Subject: [PATCH 3/4] fix soft-delete for 1.0.1

Signed-off-by: apporc <appleorchard2000@gmail.com>
Signed-off-by: fabian4 <cybing4@gmail.com>
---
 nova/volume/cinder.py | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/nova/volume/cinder.py b/nova/volume/cinder.py
index 2070fa2..8a3423f 100644
--- a/nova/volume/cinder.py
+++ b/nova/volume/cinder.py
@@ -20,6 +20,8 @@ Handles all requests relating to volumes + cinder.
 
 import copy
 import sys
+from nova import context as nova_context
+from keystoneclient.v2_0 import client as keystone_client
 
 from cinderclient import client as cinder_client
 from cinderclient import exceptions as cinder_exception
@@ -76,6 +78,17 @@ cinder_opts = [
                      'availability zones.',
                 deprecated_group='DEFAULT',
                 deprecated_name='cinder_cross_az_attach'),
+    cfg.StrOpt('admin_auth_url',
+               default='http://localhost:5000/v2.0',
+               help='Authorization URL for connecting to cinder in admin'
+                    'context.'),
+    cfg.StrOpt('admin_username',
+               help='Username for connecting to cinder in admin context'),
+    cfg.StrOpt('admin_password',
+               help='Password for connecting to cinder in admin context',
+               secret=True),
+    cfg.StrOpt('admin_tenant_name',
+               help='Tenant name for connecting to cinder in admin context.')
 ]
 
 CONF = cfg.CONF
@@ -89,6 +102,24 @@ CINDER_URL = None
 
 def cinderclient(context):
     global CINDER_URL
+
+    if context.is_admin and not context.auth_token:
+        kc = keystone_client.Client(
+            username=CONF.cinder.admin_username,
+            password=CONF.cinder.admin_password,
+            tenant_name=CONF.cinder.admin_tenant_name,
+            auth_url=CONF.cinder.admin_auth_url)
+        context = nova_context.RequestContext(
+            user_id=kc.user_id,
+            is_admin=True,
+            project_id=kc.project_id,
+            user_name=kc.username,
+            project_name=kc.project_name,
+            roles=['admin'],
+            auth_token=kc.auth_token,
+            request_id=context.request_id,
+            service_catalog=kc.service_catalog.catalog['serviceCatalog'])
+
     version = get_cinder_client_version(context)
     c = cinder_client.Client(version,
                              context.user_id,
-- 
2.1.0

