From 98526147b856cbf48ebd26ee0bd91ecbc06fd1fd Mon Sep 17 00:00:00 2001
From: Zhao Chao <zhaochao1984@gmail.com>
Date: Tue, 14 Mar 2017 22:21:23 +0800
Subject: [PATCH] Add support of update Cinder volume QoS Specs.

Signed-off-by: Zhao Chao <zhaochao1984@gmail.com>
---
 novaclient/v1_1/volumes.py | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/novaclient/v1_1/volumes.py b/novaclient/v1_1/volumes.py
index 77746dd3..d3292bda 100644
--- a/novaclient/v1_1/volumes.py
+++ b/novaclient/v1_1/volumes.py
@@ -132,6 +132,19 @@ class VolumeManager(base.ManagerWithFind):
         return self._update("/servers/%s/os-volume_attachments/%s" %
             (server_id, attachment_id,), body, "volumeAttachment")
 
+    def update_volume_qos(self, server_id, volume_id, qos_specs):
+        """
+        Update Cinder QoS Specs of the volume identified by the Volume ID,
+        that is attached to the given server ID
+
+        :param server_id: The ID of the server
+        :param volume_id: The ID of the volume
+        :param qos_specs: New Cinder QoS Specs of the volume
+        """
+        body = {'volumeQoSSpecs': qos_specs}
+        return self._update("/servers/%s/os-volume_attachments/%s" %
+            (server_id, volume_id,), body)
+
     def get_server_volume(self, server_id, attachment_id):
         """
         Get the volume identified by the attachment ID, that is attached to
-- 
2.11.0

